Automattic/woocommerce-payments
# Handle rate limit errors during the checkout process

Fixes #4801 

#### Changes proposed in this Pull Request

During the checkout process, sometimes Stripe API return an error with code that says that request is rate limited. In most cases, the payment is already processed and the only step that needs to be done is to redirect the customer to the order received page.

To address this issue, additional checks have been added that will monitor error code and if the error code is the one that tells us that rate limit is in place, an additional request will be made that will get payment intent. If payment intent is succeeded, the user will be redirected. If that's not a case, the original error message will be shown





#### Testing instructions

This flow will be difficult to test because it will be difficult to replicate stripe response with specific error code. One way to test it is to manually edit the code with a specific error code that is mentioned in the original issue. Make sure that payment intent succeeded and make sure that part of code that calls additional payment intent API request is called by manually setting error variable.

-------------------


**Post merge**
